### ISSUE TITLE
Can not create C# script type plugins, Cyclical issue

### ISSUE THREAD

**Issue by HexBlit** [godotengine/godot#87930]
### Tested versions

This is reproducible on Godot v4.2.1.stable.mono

### System information

Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (NVIDIA; 31.0.15.4633) - 13th Gen Intel(R) Core(TM) i9-13900KS (32 Threads)

### Issue description

If you try to create a C# editor plugin in v4.2.1 mono using C# as the language it shows a cyclic error when the default new plugin files are generated after going through the create new plugin workflow.
<img width="1922" alt="Screenshot 2024-02-04 001344" src="https://github.com/godotengine/godot/assets/1005054/1bf22c59-4dc6-4d2a-bdbd-5566e6c8680c">

### Steps to reproduce

In the Mono Godot editor create a new blank project. Then go and create a new plugin, and set the language C# instead of gdscript. 

A cyclic error will appear and prevent the plugin from being enabled. 

### Minimal reproduction project (MRP)

[super_highlight_pro.zip](https://github.com/godotengine/godot/files/14155852/super_highlight_pro.zip)

**Comment by AThousandShips** [comment]
Did you uncheck the "activate now" checkbox in the plugin configuration?

**Comment by HexBlit** [comment]
I did not, it does that when you create new one with it auto activating. and gives the error anytime you try to activate it.

This is a duplicate of https://github.com/godotengine/godot/issues/84878 when I checked that one, I had to make sure I compiled the csharp script per the comments in that other issue.

There should be some polish on this to not auto activate on a C# script plugin, or run a compile task before activating.
